src/scimlfunctions.jl

```diff
@@ -4308,3 +4308,82 @@ for S in [:ODEFunction
         end
     end
 end
+
+function SymbolicIndexingInterface.is_variable(fn::AbstractSciMLFunction, sym)
+  has_sys(fn) && return is_variable(fn.sys, sym)
+  has_syms(fn) && return Symbol(sym) in fn.syms
```

Comment:
Any chance we can get rid of this part? i.e. in the `SciMLFunction` constructor we could turn `syms`/`paramsyms` into a `sys` with the appropriate behavior.